src/assign.c

```diff
@@ -173,21 +172,20 @@ static SEXP shallow(SEXP dt, SEXP cols, R_len_t n)
 
   SEXP names = PROTECT(getAttrib(dt, R_NamesSymbol)); protecti++;
   SEXP newnames = PROTECT(allocVector(STRSXP, n)); protecti++;
+  const int l = isNull(cols) ? LENGTH(dt) : length(cols);
   if (isNull(cols)) {
-    l = LENGTH(dt);
-    for (i=0; i<l; i++) SET_VECTOR_ELT(newdt, i, VECTOR_ELT(dt,i));
+    for (int i=0; i<l; ++i) SET_VECTOR_ELT(newdt, i, VECTOR_ELT(dt,i));
```

Comment:
why change from R_len_t to int?

Reply:
I'm kind of fed up with `R_len_t`. It's point (I assume, because it was a good idea) was so that R could change its definition from `int` to `int64_t` (for example) under the covers without affecting package space. But that wasn't the direction taken by R (possibly because use of `R_len_t` wasn't/couldn't be enforced so lots of `int` existed in package space which would have broken if `R_len_t` was changed from `int` to `int64_t`) and we have to make that change anyway and look for the one letter difference between `R_len_t` and `R_xlen_t`. We're using `int` more often now than `R_len_t`, and certainly `int64_t` more than `R_xlen_t`. When I tried last time to change everything to int64_t, my brain and greps found it much easier looking at `int`.

Reply:
I see, I had been thinking (based on pure speculation) `R_len_t` and `R_xlen_t` were meant to be there, but that their point was platform robustness (akin to `int_fast32_t` and so on). Glad to know your reasoning.